packages/playback-core/src/index.ts

```diff
@@ -23,173 +21,28 @@ export const generatePlayerInitTime = () => {
-  /** @TODO Figure out a way to "downgrade" derived types below to early TS syntax (e.g. 3.4) instead of explicit versions here (CJP) */
-  'hls',
-  'HLS',
-  // ...(shorthandKeys as (keyof MimeTypeShorthandMap)[]),
-  // ...(shorthandKeys.map((k) => k.toUpperCase()) as `${Uppercase<keyof MimeTypeShorthandMap>}`[]),
-  // ...(shorthandKeys.map((k) => k.toLowerCase()) as `${Lowercase<keyof MimeTypeShorthandMap>}`[]),
-] as MediaTypes[];
-
-export type MuxMediaPropTypes = {
-  envKey: Options['data']['env_key'];
-  debug: Options['debug'] & Hls['config']['debug'];
-  metadata: Partial<Options['data']>;
-  customDomain: string;
-  beaconCollectionDomain: Options['beaconCollectionDomain'];
-  errorTranslator: Options['errorTranslator'];
-  playbackId: string;
-  playerInitTime: Options['data']['player_init_time'];
-  preferPlayback: ValueOf<PlaybackTypes> | undefined;
-  type: MediaTypes;
-  streamType: ValueOf<StreamTypes>;
-  startTime: HlsConfig['startPosition'];
-  autoPlay: boolean | ValueOf<AutoplayTypes>;
-  autoplay: boolean | ValueOf<AutoplayTypes>;
-};
-
-export type HTMLMediaElementProps = Partial<Pick<HTMLMediaElement, 'src'>>;
-
-export type MuxMediaProps = HTMLMediaElementProps & MuxMediaPropTypes;
-export type MuxMediaPropsInternal = MuxMediaProps & {
-  playerSoftwareName: Options['data']['player_software_name'];
-  playerSoftwareVersion: Options['data']['player_software_version'];
-};
-
-export const toPlaybackIdParts = (playbackIdWithOptionalParams: string): [string, string?] => {
-  const qIndex = playbackIdWithOptionalParams.indexOf('?');
-  if (qIndex < 0) return [playbackIdWithOptionalParams];
-  const idPart = playbackIdWithOptionalParams.slice(0, qIndex);
-  const queryPart = playbackIdWithOptionalParams.slice(qIndex);
-  return [idPart, queryPart];
-};
-
 export const toMuxVideoURL = (playbackId?: string, { domain = MUX_VIDEO_DOMAIN } = {}) => {
   if (!playbackId) return undefined;
   const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
   return `https://stream.${domain}/${idPart}.m3u8${queryPart}`;
 };
 
-export const inferMimeTypeFromURL = (url: string) => {
-  let pathname = '';
-  try {
-    pathname = new URL(url).pathname;
-  } catch (e) {
-    console.error('invalid url');
-  }
-
-  const extDelimIdx = pathname.lastIndexOf('.');
-  if (extDelimIdx < 0) return '';
-
-  const ext = pathname.slice(extDelimIdx + 1);
-  const upperExt = ext.toUpperCase();
-
-  return isKeyOf(upperExt, ExtensionMimeTypeMap) ? ExtensionMimeTypeMap[upperExt] : '';
-};
-
-export const getType = (props: Partial<Pick<MuxMediaProps, 'type' | 'src'>>) => {
-  const type = props.type;
-
-  if (type) {
-    const upperType = type.toUpperCase();
-
-    return isKeyOf(upperType, MimeTypeShorthandMap) ? MimeTypeShorthandMap[upperType] : type;
-  }
-
-  const { src } = props;
-
-  if (!src) return '';
-
-  return inferMimeTypeFromURL(src);
-};
-
-export const getStreamTypeConfig = (streamType?: ValueOf<StreamTypes>) => {
```

Comment:
non-blocking: I kind of think this should stay in this file because this method defines the default configs. Though, I won't insist on this if you'd prefer otherwise.

Reply:
good call, will move